### PR TITLE
Run chmod +x explicitly in stage0

### PR DESCRIPTION
**Pull request by eed3si9n**
This is on top of https://github.com/sbt/sbt-native-packager/pull/1200

Using sbt-native-packager 1.3.17 with OpenShift, I still ran into some permission issues and needed to call chmod +x explicitly in the container. This will emit chmod +x for all bash scripts that was generated.

```Dockerfile
RUN ["chmod", "u=+x,g=+x", "/opt/docker/bin/file-permission-test"]
```

### CODE REVIEW

File: src/main/scala/com/typesafe/sbt/packager/docker/DockerPermissionStrategy.scala [hunk 1]
```diff
@@ -61,6 +68,13 @@ object DockerChmodType {
     def argument: String = "u=rwX,g=rwX"
   }
 
+  /**
+    * Gives +x permission to users and groups.
+    */
+  case object UserGroupPlusExecute extends DockerChmodType {
+    def argument: String = "u=+x,g=+x"
```
Comment (eed3si9n) [inline src/main/scala/com/typesafe/sbt/packager/docker/DockerPermissionStrategy.scala]: This is a mistake. It should be u+x,g+x.

**Comment by eed3si9n** [comment]
@muuki88 Fixed a bug that the tests caught, and also updated the docs.

File: src/sphinx/formats/docker.rst [hunk 2]
```diff
@@ -128,11 +128,16 @@ Environment Settings
...
-    * ``DockerChmodType.SyncGroupToUser``: chmod -R g=u
+    * ``DockerChmodType.UserGroupReadExecute`` (default): chmod u=rX,g=rX
+    * ``DockerChmodType.UserGroupRead``: chmod u=r,g=r
+    * ``DockerChmodType.UserGroupWriteExecute``: chmod u=rwX,g=rwX
+    * ``DockerChmodType.SyncGroupToUser``: chmod g=u
+    * ``DockerChmodType.UserGroupPlusExecute``: chmod u+x,g+x (This is for ``dockerAdditionalPermissions``)
     * ``DockerChmodType.Custom``: Custom argument provided by the user.
 
+  ``dockerAdditionalPermissions``
+    Additional permissions typically used to give ``chmod +x`` rights for the executable files. By default generated Bash scripts are given ``DockerChmodType``.
```
Comment (olivierlemasle) [inline src/sphinx/formats/docker.rst]: Missing permissions?

Comment (muuki88) [reply src/sphinx/formats/docker.rst]: Added this via the UI :+1: thanks for reviewing this as well :hug

File: src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala [hunk 3]
```diff
@@ -108,6 +108,16 @@ object DockerPlugin extends AutoPlugin {
     },
     dockerRmiCommand := dockerExecCommand.value ++ Seq("rmi"),
     dockerBuildCommand := dockerExecCommand.value ++ Seq("build") ++ dockerBuildOptions.value ++ Seq("."),
+    dockerAdditionalPermissions := {
+      val scripts = makeBashScripts.value
```
Comment (muuki88) [inline src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala]: I overlooked that. This causes a circular dependency between `JavaAppPackaging` and the `DockerPlugin`, which produces a bug reported in #1206

**Pull request merged**